src/jit/gentree.cpp

```diff
@@ -16869,14 +16867,15 @@ bool FieldSeqNode::IsPseudoField()
 GenTreeSIMD* Compiler::gtNewSIMDNode(
     var_types type, GenTreePtr op1, SIMDIntrinsicID simdIntrinsicID, var_types baseType, unsigned size)
 {
-    // TODO-CQ: An operand may be a GT_OBJ(GT_ADDR(GT_LCL_VAR))), in which case it should be
-    // marked lvUsedInSIMDIntrinsic.
     assert(op1 != nullptr);
     if (op1->OperGet() == GT_LCL_VAR)
     {
-        unsigned   lclNum                = op1->AsLclVarCommon()->GetLclNum();
-        LclVarDsc* lclVarDsc             = &lvaTable[lclNum];
-        lclVarDsc->lvUsedInSIMDIntrinsic = true;
+        setLclRelatedToSIMDIntrinsic(op1);
+    }
+    else if ((op1->OperGet() == GT_OBJ) && (op1->gtOp.gtOp1->OperGet() == GT_ADDR) &&
+             op1->gtOp.gtOp1->gtOp.gtOp1->OperIsLocal())
+    {
+        setLclRelatedToSIMDIntrinsic(op1->gtOp.gtOp1->gtOp.gtOp1);
     }
```

Comment:
Looks like this logic is repeated 3 times.  How about abstracting it in a routine SetLclRelatedToSMDIntrinsic()?

The caller would have to call

SetLclRelatedToSMDIntrinsic(op1);
if (op2 != nullptr)
{
   SetLclRelatedToSMDIntrinsic(op2);
}